File: Safrel/src/main/java/performancetest/VirtualMachine.java
package performancetest;


public class VirtualMachine {
    public double Throughput;
    public double ResponseTime_i;
    public double ResponseTime;
    public double NormalizedResponsetime;
    public double Requirement_ResTime;
    public double Acceptolerance;
    // 4 cores with 2.5 GHz
    public int VM_CPU_g;
    public long VM_Mem_g;
    public long VM_Disk_g;
    public int VM_CPU_i;
    public long VM_Mem_i;
    public long VM_Disk_i;
    public double VM_CPUtil;
    public double VM_MemUtil;
    public double VM_DiskUtil;
    public Double[] VM_SensitivityValues;


    public void CalculateCPUtilImprov() {
        this.VM_CPUtil= (double) this.VM_CPU_g /this.VM_CPU_i;
    }
    
    public void CalculateMemUtilImprov() {
        this.VM_MemUtil= (double) this.VM_Mem_g /this.VM_Mem_i;
    }
    
    public void CalculateDiskUtilImprov() {
        this.VM_DiskUtil= (double) this.VM_Disk_g /this.VM_Disk_i;
    }
    
    public void NormalizeResponseTime() {
        // R_Norm= b.2/Math.Pi
        this.NormalizedResponsetime= (2.0/Math.PI) * Math.atan(((this.ResponseTime/this.Requirement_ResTime)));
    }
}

File: ppo/src/rl_load_tester/src/virtualMachine.py
import os


class VirtualMachine:
    def __init__(self, Requirement_ResTime):
        self.VM_CPU_i = 0
        self.VM_Mem_i = 0.0
        self.VM_CPU_g = self.VM_CPU_i
        self.VM_Mem_g = self.VM_Mem_i
        self.ResponseTime_i = 0
        self.Requirement_ResTime = Requirement_ResTime
        self.ResponseTime = 0

        self.threads = int(os.getenv("THREADS", 2))
        self.rampup = int(os.getenv("RAMPUP", 1))
        self.loops = int(os.getenv("LOOPS", 1))

    def reset(self):
        self.VM_CPU_i = 90
        self.VM_Mem_i = 1.8
        self.VM_CPU_g = self.VM_CPU_i
        self.VM_Mem_g = self.VM_Mem_i
        self.ResponseTime_i = 0
        self.ResponseTime = 0

        self.threads = int(os.getenv("THREADS", 2))
        self.rampup = int(os.getenv("RAMPUP", 1))
        self.loops = int(os.getenv("LOOPS", 1))
